### ISSUE TITLE
glibc: cannot compute suffix of object files: cannot compile

### ISSUE THREAD

**Issue by fuzihaofzh** [Linuxbrew/brew#43]
# Please follow the general troubleshooting steps first:
- [ ] Ran `brew update` and retried your prior step?
- [ ] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ ] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

_You can erase any parts of this template not applicable to your Issue._
### Bug reports:

I am trying to install tmux, but I got:

```
brew install tmux
==> Installing dependencies for tmux: glibc, xz, gmp, gpatch, mpfr, libmp
==> Installing tmux dependency: glibc
==> Downloading http://ftpmirror.gnu.org/glibc/glibc-2.19.tar.bz2
Already downloaded: /home/zihao.fzh/.cache/Homebrew/glibc-2.19.tar.bz2
==> ../configure --disable-debug --disable-dependency-tracking --disable-silent-
Last 15 lines from /home/zihao.fzh/.cache/Homebrew/Logs/glibc/01.configure:
--disable-debug
--disable-dependency-tracking
--disable-silent-rules
--prefix=/home/zihao.fzh/.linuxbrew/Cellar/glibc/2.19
--enable-obsolete-rpc
--without-selinux
--with-binutils=/home/zihao.fzh/.linuxbrew/Cellar/binutils/2.26/bin
--with-headers=/home/zihao.fzh/.linuxbrew/Cellar/linux-headers/3.18.27/include

checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking for gcc... /usr/bin/gcc
checking for suffix of object files... configure: error: in `/tmp/glibc-20160519-8022-1ipic1m/glibc-2.19/build':
configure: error: cannot compute suffix of object files: cannot compile
See `config.log' for more details
```

**Comment by DoomHammer** [comment]
That's strange. Why does `tmux` require `glibc`, @sjackman ?

**Comment by sjackman** [comment]
All bottles depend on `glibc` and `gcc` (on systems with host `glibc` < 2.19).

**Comment by sjackman** [comment]
@maplewizard Please report the output of `brew gist-logs glibc`

**Comment by fuzihaofzh** [comment]
@sjackman 
The log are: https://gist.github.com/a38d873e11c3084727cb0a7cf8413af1
I found that brew will use system g++(4.1.2) while my personal g++(4.9.3) will not be used. So I tried 
`brew install glibc --cc=gcc`
while it cannot use my personal gcc neither.
The new logs are: https://gist.github.com/anonymous/0efc130ffb9755142c2b53aacab4f1aa
when I input `gcc -v`. It shows that `gcc version 4.9.3 (GCC)`
How can I force brew to use my personal gcc? `--cc=gcc` seems not work.
